File: SharedLibrary/Shared/ApiModels/Student.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;
using SharedLibrary.Interfaces;

namespace SharedLibrary.Shared.ApiModels
{
	public class Student : IStudent
	{
		public int ID { get; set; }
		public int? ProfilePhotoID { get; set; }
		public int? ClassID { get; set; }
		public string Name { get; set; }
		public string Lastname { get; set; }
		public string ClassName { get; set; }
		public int ZolikCount { get; set; }
		public int XP { get; set; }

		[JsonIgnore]
		public string FullName => $"{Name} {Lastname}";

		public Student() { }

		public Student(IStudent from)
		{
			this.ID = from.ID;
			this.ProfilePhotoID = from.ProfilePhotoID;
			this.ClassID = from.ClassID;
			this.Name = from.Name;
			this.Lastname = from.Lastname;
			this.ClassName = from.ClassName;
		}
	}

	public class Student<T> : Student, IStudent<T> where T : class, IImage, new()
	{
		public T GetDefaultImage() => new T {
			ID = 4,
			OwnerID = 2,
			Base64 =
				"iVBORw0KGgoAAAANSUhEUgAAAMAAAADACAAAAAB3tzPbAAACOUlEQVR4Ae3aCQrrIBRG4e5/Tz+CBAlIkIAECUjoSt48z/GZeAvnrMCvc6/38XzxAAAAYC4AAAAAAAAAAAAAAAAAAAAAAAAAAAAMCAAAAAAAAAAAAAAAAPsagz4V4rq/FmCLTj/k4vYqgCN5/TKfjlcAJKff5pJ5QPH6Y77YBiz6a4thQJ30D03VKmB3+qfcbhOwO+l+waP/+VsEBgDV6USumgNMOtVkDbDoZIstQNHpiimA1+m8JUBSQ8kO4HBqyB1mAElNJTMAr6a8FcCmxjYjgKjGohGAU2POBmBXc7sJwKrmVhOAqOaiCUBQc8EEQO0JwPMB4ADASwhAe3yR8VPiP3/M8XOaPzQd/lLyp56xSuvnUGK0yHC313idCw6umNov+bhm5aK7fdWAZQ/WbdoXnlg5Y+mvfe2SxVdWj20FAAAAAAAAAAAAwFQAAJSS0hwmfVMIc0qlmAfsOQWvP+RDyrtNQM1L0D8WllxNAWqOXifzMVcbgG3xaswv22jAFp3a6zFteYw8fQ9DM6Amr275VG8GlFmdm8uNgDzpgqZ8EyB7XZTPNwDKpAubysWAOuvi5nolYHW6PLdeBjiCbikc1wCK0025cgUg68Zyf0DUrcXegKibi30Bq25v7QnYNKCtH+BwGpA7ugFmDWnuBSgaVOkECBpU6AOoGlbtAlg1rLULIGhYoQvAaViuC0AD6wE4Xh1QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADA194CuqC6onikxXwAAAAASUVORK5CYII=",
			MIME = "image/png",
			Size = 626
		};

		public T ProfilePhoto { get; set; }

		public T Profile
		{
			get => ProfilePhoto ?? this.GetDefaultImage();
			set
			{
				if (value == null) {
					value = this.GetDefaultImage();
				}
				ProfilePhoto = value;
			}
		}

		public Student() { }

		public Student(IStudent<T> from) : base(from)
		{
			this.ProfilePhoto = from.ProfilePhoto;
		}
	}
}

File: MobileApp/zoliky_teachers/lib/utils/api/models/Student.dart
import 'package:zoliky_teachers/utils/api/models/Image.dart';
import 'package:zoliky_teachers/utils/interfaces/IDbObject.dart';

class Student implements IDbObject {
  int id;
  int profilePhotoId;
  int classId;
  String name;
  String lastname;
  String className;
  int zolikCount;
  int xp;

  Image profilePhoto;

  String get fullName => this.name + " " + this.lastname;

  Student();

  static Student fromJson(Map<String, dynamic> json) {
    Student u = new Student();
    if (json.containsKey("ID") && json["ID"] != null) {
      u.id = json["ID"];
    }
    if (json.containsKey("ProfilePhotoID") && json["ProfilePhotoID"] != null) {
      u.profilePhotoId = json["ProfilePhotoID"];
    }
    if (json.containsKey("ClassID") && json["ClassID"] != null) {
      u.classId = json["ClassID"];
    }
    if (json.containsKey("Name") && json["Name"] != null) {
      u.name = json["Name"];
    }
    if (json.containsKey("Lastname") && json["Lastname"] != null) {
      u.lastname = json["Lastname"];
    }
    if (json.containsKey("ClassName") && json["ClassName"] != null) {
      u.className = json["ClassName"];
    }
    if (json.containsKey("ZolikCount") && json["ZolikCount"] != null) {
      u.zolikCount = json["ZolikCount"];
    }
    if (json.containsKey("XP") && json["XP"] != null) {
      u.xp = json["XP"];
    }
    if (json.containsKey("ProfilePhoto") && json["ProfilePhoto"] != null) {
      u.profilePhoto = Image.fromJson(json["ProfilePhoto"]);
    } else {
      u.profilePhoto = Image.defaultImage();
    }
    return u;
  }
}
